transcom/mymove
# B-20462 Enforce N9 1W Character Limit INT

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A979801)

## Summary

We are needing to enforce some characters limits in the EDI 858 invoice generation for fields that have caused issues or could cause issues in the future. One of those fields is the Service Member Name, the N9 1W segment of the EDI 858. We want to truncate the middle name down to the middle initial only, and truncate the entire name down to 30 characters if it happens to be longer. The truncation will also end the name with three trailing dots `...` to indicate that it's been truncated.

To test:

1. Click on View Testharness Data Scenarios as an office user
![image](https://github.com/user-attachments/assets/5db884e3-2f68-44d8-a2fd-53cc255e62d4)
2. Click on MakeMoveReadyForEDI
3. Search for the first_name that is generated, and find that in the database. You'll also use the payment_request_number from this output so don't close out of it.
4. Modify the first name, middle name, and last name to be whatever you want that is under 30 characters total for lastName, M. firstName
5. Generate the EDI 858 locally - `go run ./cmd/generate-payment-request-edi/ --payment-request-number <payment_request_number>` where the payment_request_number comes from the MakeMoveReadyForEDI output.
6. Search for 1W in the output, and check that the middle name is shortened to just the middle initial.
7. Modify the first name, middle name, and last name to be something longer than 30 characters total.
8. Regenerate the EDI 858 and search for 1W in the output, and check that it is truncated down to 30 characters and ends in `...`

## Screenshots
Truncating middle name down to middle initial:
![image](https://github.com/user-attachments/assets/b3800b77-8d39-493b-956a-704da41a517b)

- EDI 858 generated:

![image](https://github.com/user-attachments/assets/5c9c170e-f91d-43f6-8c48-66d6497bc1ad)

Truncating the entire name down to 30 characters:
![image](https://github.com/user-attachments/assets/bde6c6e0-1e2b-4da8-aa25-adaee610b3c8)

- EDI 858 generated:

![image](https://github.com/user-attachments/assets/d5935614-51b2-477f-9d52-9d755c256d26)
